bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/ReadCache.java

```diff
@@ -90,6 +93,10 @@ public void put(long ledgerId, long entryId, ByteBuf entry) {
         lock.readLock().lock();
 
         try {
+            if (entrySize > segmentSize) {
+                log.warn("entrySize {} > segmentSize {}, skip update read cache!", entrySize, segmentSize);
+                return;
+            }
             int offset = currentSegmentOffset.getAndAdd(alignedSize);
             if (offset + entrySize > segmentSize) {
                 // Roll-over the segment (outside the read-lock)
```

Comment:
If you must return it can be returned from here, instead of rewriting this segment.
 if (entrySize > segmentSize) {
                log.warn("entrySize {} > segmentSize {}, skip update read cache!", entrySize, segmentSize);
                return;
            }

Reply:
Never mind. I misread the code.